vignettes/queries_dataset.Rmd

```diff
@@ -0,0 +1,179 @@
+    
+    Different `TERM_LEVEL` variables may be specified within a `VAR_PREFIX`.
+
+  + `TERM_NAME` must be a character string. 
+  This **must** be populated if `TERM_ID` is missing.
+  
+  + `TERM_ID` must be an integer. 
+  This **must** be populated if `TERM_NAME` is missing.
+  
+
+
+### Output
+
+  + `VAR_PREFIX` will be used to create the grouping variable appending the suffix "NAM". This variable will now be referred to as `ABCzzNAM`: the name of the grouping variable.
+
+    E.g. `VAR_PREFIX == "SMQ01"` will create the `SMQ01NAM` variable.
+
+    
+
+
+    For each `VAR_PREFIX`, a new `ABCzzNAM` variable is created in `dataset`.
+
+
+  + [`QUERY_NAME`]{#query_name} will be used to populate the corresponding `ABCzzNAM` variable.
+
+  + `TERM_LEVEL` will be used to identify the variables from `dataset` used to perform the grouping (e.g. `AEDECOD`,`AEBODSYS`, `AELLTCD`).
+
+  + `TERM_NAME` (for character variables), `TERM_ID` (for numeric variables) will be used to identify the records meeting the criteria in `dataset` based on the variable defined in `TERM_LEVEL`.
+  
+  
+  + **Result:**
+  
+    For each record in `dataset`, where the variable defined by `TERM_LEVEL` match a term from the `TERM_NAME` (for character variables) or `TERM_ID` (for numeric variables) in the `datasets_queries`, `ABCzzNAM` is populated with `QUERY_NAME`.
+    
+    
+    Note: The type (numeric or character) of the variable defined in `TERM_LEVEL` is checked in `dataset`. If the variable is a character variable (e.g. `AEDECOD`), it is expected that `TERM_NAME` is populated, if it is a numeric variable (e.g. `AEBDSYCD`), it is expected that `TERM_ID` is populated, otherwise an error is issued.
+    
+
+### Example
+
+In this example, 1 standard MedDRA query (`VAR_PREFIX = "SMQ01"`) and 1 customized query (`VAR_PREFIX = "CQ02"`) are  defined to analyze the adverse events. 
+
+  + The standard MedDRA query variable `SMQ01NAM` [`VAR_PREFIX`] will be populated with "Standard Query 1" [`QUERY_NAME`] if any preferred term (`AEDECOD`) [`TERM_LEVEL`] in `dataset` is equal to "AE1" or "AE2" [`TERM_NAME`]
+  
+  + The customized query (`CQ02NAM`) [`VAR_PREFIX`] will be populated with "Query 2" [`QUERY_NAME`] if any Low Level Term Code (`AELLTCD`)  [`TERM_LEVEL`] in `dataset` is equal to 10  [`TERM_ID`] or any preferred term (`AEDECOD`) [`TERM_LEVEL`] in `dataset` is equal to "AE4" [`TERM_NAME`].
+  
+  **`dataset_queries = ds_query`: **
+  
+VAR_PREFIX | QUERY_NAME | TERM_LEVEL | TERM_NAME | TERM_ID |
+------- | ----- | ------ | ----- | ----- | -----
+SMQ01| Standard Query 1 | AEDECOD | AE1 | 
+SMQ01| Standard Query 1 | AEDECOD | AE2 | 
+CQ02| Query 2 | AELLTCD | | 10
+CQ02| Query 2 | AEDECOD | AE4| 
+
+  **`dataset = ae`: **
+  
+USUBJID | AEDECOD | AELLTCD 
+------- | ----- | ------ 
+0001 | AE1 | 101 | 
+0001 | AE3 | 10| | 
+0001 | AE4 | 120 | 
+0001 | AE5 | 130 |
+
+
+  **output dataset from `derive_vars_query(dataset = ae, dataset_queries = ds_query)`:**
```

Comment:
Can we use "proper" headng here instead of just bold text? I would suggest:

```
#### Query Dataset (`ds_query`)

#### Adverse Event Dataset (`ae`)

#### Output Dataset

Generated by calling  `derive_vars_query(dataset = ae, dataset_queries = ds_query)`.
```

Reply:
updated as suggested